src/credentials/FabricTable.h

```diff
@@ -386,11 +386,11 @@ class DLL_EXPORT FabricTable
         virtual void OnFabricCommitted(const FabricTable & fabricTable, FabricIndex fabricIndex){};
 
         /**
-         * Gets called when operational credentials are changed, which may not be persistent.
+         * Gets called when operational credentials are added or updated, which may not be persistent.
          *
-         * Can be used to affect what is needed for UpdateNOC prior to commit.
+         * Can be used to affect what is needed for UpdateNOC prior to commit for fabric updating.
          **/
-        virtual void OnFabricUpdated(const FabricTable & fabricTable, FabricIndex fabricIndex){};
+        virtual void OnFabricAddedOrUpdated(const FabricTable & fabricTable, FabricIndex fabricIndex, bool isAddition){};
```

Comment:
This needs to be updated to be an *additional* callback. The Updated callback MUST also be called.